File: src/main/java/io/github/falon/scryingstatues/item/ShattererItem.java
package io.github.falon.scryingstatues.item;

import io.github.falon.scryingstatues.ScryingStatues;
import io.github.falon.scryingstatues.entity.statue.StatueEntity;
import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ActionResult;
import net.minecraft.util.Hand;

public class ShattererItem extends Item {
	public ShattererItem(Settings settings) {
		super(settings);
	}

	@Override
	public ActionResult useOnEntity(ItemStack stack, PlayerEntity user, LivingEntity entity, Hand hand) {
		ScryingStatues.LOGGER.info("Shattering entity: {}", entity.toString());

		return ActionResult.SUCCESS;
	}
}

File: src/main/java/io/github/falon/scryingstatues/item/ShattererItem.kt
package io.github.falon.scryingstatues.item

import net.minecraft.entity.LivingEntity
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.util.ActionResult
import net.minecraft.util.Hand

class ShattererItem(settings: Settings?) : Item(settings) {
    override fun useOnEntity(stack: ItemStack, user: PlayerEntity, entity: LivingEntity, hand: Hand): ActionResult {
        return ActionResult.SUCCESS
    }
}
